src/js/wrapper.ts

```diff
@@ -40,16 +40,28 @@ export const NATIVE = {
       sdk: event.sdk,
     };
 
-    const payload = {
-      ...event,
-      message: {
-        message: event.message,
-      },
-    };
-
     if (NATIVE.platform === "android") {
       const headerString = JSON.stringify(header);
 
+      /*
+        We do this to avoid duplicate breadcrumbs on Android as sentry-android applies the breadcrumbs
+        from the native scope onto every envelope sent through it. This scope will contain the breadcrumbs
+        sent through the scope sync feature. This causes duplicate breadcrumbs.
+        We then remove the breadcrumbs here but only when mechanism.handled is true. If it is handled == false,
+        this is a signal that the app would crash and android would lose the breadcrumbs by the time the app is restarted to read
+        the envelope.
+      */
+      if (event.exception?.values?.[0]?.mechanism?.handled) {
```

Comment:
Do we need to check if scope sync is enabled? And the native layer is enabled etc? IIRC we have something like `enableNative` here

Reply:
yep we agreed on that too

Reply:
@bruno-garcia @marandaneto  `enableNative` is already checked beforehand and scope sync is always enabled, we haven't added an option to disable it yet.